src/test/llm/response_parsers/hermes3_response_parser_test.cpp

```diff
@@ -0,0 +1,129 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gtest/gtest.h>
+#include <openvino/genai/tokenizer.hpp>
+#include <string>
+#include <vector>
+
+#include "../../../llm/response_parsers/base_response_parser.hpp"
+#include "../../../llm/response_parsers/response_parser.hpp"
+
+using namespace ovms;
+
+const std::string tokenizerPath = "/ovms/src/test/llm_testing/NousResearch/Hermes-3-Llama-3.1-8B";
+
+class Hermes3ResponseParserTest : public ::testing::Test {
+protected:
+    std::unique_ptr<ov::genai::Tokenizer> tokenizer;
+    std::unique_ptr<ResponseParser> responseParser;
+
+    void SetUp() override {
+        tokenizer = std::make_unique<ov::genai::Tokenizer>(tokenizerPath);
+        responseParser = std::make_unique<ResponseParser>(*tokenizer, "hermes3");
+    }
+};
+
+TEST_F(Hermes3ResponseParserTest, ParseToolCallOutputWithSingleToolCall) {
+    std::string inputWithProperClosure = "<tool_call>{\"name\": \"example_tool\", \"arguments\": {\"arg1\": \"value1\", \"arg2\": 42}}</tool_call>";
+    std::string inputWithImproperClosure = "<tool_call>{\"name\": \"example_tool\", \"arguments\": {\"arg1\": \"value1\", \"arg2\": 42}}";
+
+    // Hermes3 may produce last tool call without closing tag, so we test both cases
+    // The results should be identical
+    std::vector<std::string> inputs = {inputWithProperClosure, inputWithImproperClosure};
+    for (auto& input : inputs) {
+        auto generatedTensor = tokenizer->encode(input, ov::genai::add_special_tokens(false)).input_ids;
+        std::vector<int64_t> generatedTokens(generatedTensor.data<int64_t>(), generatedTensor.data<int64_t>() + generatedTensor.get_size());
+        ParsedResponse parsedResponse = responseParser->parse(generatedTokens);
+        EXPECT_EQ(parsedResponse.content, "");
+        EXPECT_EQ(parsedResponse.reasoning, "");
+        EXPECT_EQ(parsedResponse.reasoningTokenCount, 0);
+        EXPECT_EQ(parsedResponse.toolCalls.size(), 1);
```

Comment:
assert before accessing [0]. we already have enough of segfaults in unit tests!
please do that in all other cases and all other expects that then continue with undefined behavior in negative scenario